platform/api.search/src/org/netbeans/modules/search/BasicSearchForm.java

```diff
@@ -1119,8 +1121,23 @@ private void limitReached() {
         public MultiLineComboBoxEditor() {
             area.setWrapStyleWord(false);
             area.setLineWrap(false);
+
+            // retain standard focus traversal behavior
             area.setFocusTraversalKeys(KeyboardFocusManager.FORWARD_TRAVERS‌​AL_KEYS, null);
             area.setFocusTraversalKeys(KeyboardFocusManager.BACKWARD_TRAVERS‌​AL_KEYS, null);
+
+            // pretend to be a single line text field if you are small enough
+            area.addKeyListener(new KeyAdapter() {
```

Comment:
This will break accessibility. As a developer you know, that the behavior, but a user will not know and will be surprised if the field suddenly changes bahaviour. How about a toggle button, that switches between single and multiline case? Activating that could even provide a textarea, that is large enough by default to show multiple lines of text.

Reply:
it is intended to be intuitive without the need to explain or configure anything. If it looks like a text field, it behaves just like it did before (users seem to like to use the Enter key as OK dialog action according to the mailing list and issue). But if you have a multi line-string in it you expect it having a working Enter key for new lines - if it would trigger the dialog action it would look like a bug too.

toggle would create more issues. since if you turn it off again the Enter key would suddenly be the dialog action and the text field would contain multi line strings.

Reply:
Still **I** would not guess, that I have to resize the dialog to do multiline searches and this has zero discoverability. The dialog is big enough, so why would I resize it?

Also think about people with bad eyesight, using assistive technologies, they will even be worse, because not only have they use their tech, the UI suddenly changes behavior, just because it was scaled.

For the toggle button. Toggling it can remove newlines or turn the search to regexp based search, that will happily serach for newlines even before this change. It would also be clearer, because a button can be expected to change behavior, resizing in my mind can't.

Reply:
I agree on the discoverability aspect but this PR does not change anything there. There is already a msg label somewhere e.g stating "the text will be used literally". It could also add "resize window for multi line Strings". Adding a button which makes the window larger is unusual UI-wise, but this could be added too if there is a need (I would prefer not).

there are many little details to consider here. e.g if you select text in the editor, the dialog will already have properly sized texfields. The text fields are also used in the regex tester etc - so we would need to add those toggle buttons everywhere.

This is a very trivial issue IMO. The worst thing which can happen is that someone adds a newline in a text field instead of triggering the OK action - and only if the text field has focus. I wouldn't have expected that anyone would file an issue for that tbh.

This PR fixes it and restores original behavior for single line search/replace.

Reply:
I agree with @matthiasblaesing that this seems worse from an accessibility / discoverability point of view.  It may seem trivial, but it's been mentioned twice already, and it's annoyed me about 10 times already, and I know to press `CTRL+ENTER`.  Keyboard control of dialogs is second nature for some people!

I think either make the dialog show at least 2 lines so that multiline behaviour is expected, add some information about `CTRL-ENTER` (and whatever it is on macOS), add another way of accessing multiline, or revert and rethink after NB13.

Reply:
@neilcsmith-net please give it a try first and test it. This PR should restore the original behavior. We can improve discoverability of the multi line feature later - this has low priority for me. A hidden feature is not a bug, this PR fixes a regression caused by https://github.com/apache/netbeans/pull/3393.

just to be to be sure i repeat: this PR changes nothing regarding discoverability, it restores original behavior so that if you are used to Enter as action trigger, you can keep using it.

Reply:
with this PR applied the following should be possible again:
 - press CTRL+SHIFT+H
 - type to enter search string
 - press ENTER to trigger search
(no mouse required) 

 which should be to my knowledge EXACTLY as it was before.

Reply:
btw CTRL+ENTER also works fine - i just tested it. It works in both cases, even if the well hidden and new multi line search is active.

@neilcsmith-net Maybe i am doing something wrong, but you said "it's annoyed me about 10 times already, and I know to press CTRL+ENTER". How can i reproduce this?

Reply:
@mbien `CTRL+ENTER` works fine. You're reading my comment the opposite of how it was intended.  For those of use who've used this for decades, switching away from just `ENTER` takes some relearning of muscle memory! :smile:

How about we do the reverse here anyway? Other UIs (not NetBeans) use `SHIFT+ENTER` or `ALT+ENTER` to type newline in these sort of cases. Let's keep the default behaviour as it was.

Reply:
I tried a less invasive approach here: https://github.com/matthiasblaesing/netbeans/commit/99c6b0b878aff54c498c1ed38c54006cd48abf4b

- The renderer for previous searches replaces new lines with a single space, so that the text is still readable, but compact
- An information text is shown in the notification area, that the search can be submitted by pression CTRL+enter (might need wording adjustment and handling the mac case)
- the input area is widend to approx. 3 lines, so that text can be realistically be added. In the future it might be optimized by being able to open a dialog with a bigger TextArea
- the border of the textarea used as editor is ajusted to match the normal combo box editor one

![image](https://user-images.githubusercontent.com/2179736/150637037-965a6f45-e710-437c-b644-a15b878e4574.png)

This is not polished, but IMHO improves the situation.

Reply:
> @mbien `CTRL+ENTER` works fine. You're reading my comment the opposite of how it was intended. For those of use who've used this for decades, switching away from just `ENTER` takes some relearning of muscle memory! smile

ok understood, my mistake


> How about we do the reverse here anyway? Other UIs (not NetBeans) use `SHIFT+ENTER` or `ALT+ENTER` to type newline in these sort of cases. Let's keep the default behaviour as it was.

i get the feeling that everyone made the decision already before testing this fix, because... thats exactly what it does. It keeps the default behavior as it was :)

Reply:
@matthiasblaesing lets not redesign the dialog in a bugfix PR during rc2. The discoverability improvement can be addressed later.

Reply:
![current](https://user-images.githubusercontent.com/114367/150637667-d52596a0-bcd8-47b5-8aed-4e22880fc7a4.png)

thats the version right now after this PR. Enter works, no change in behavior form 12.6.

Reply:
> thats the version right now after this PR. Enter works, no change in behavior form 12.6.

Until you resize the dialog, then "magic" happens. The "magic" part is what I don't like. It is not discoverable and can't be observed by accessibility tools. As a user you have to "know" that something will happen or you don't and get surprised.

Reply:
> > thats the version right now after this PR. Enter works, no change in behavior form 12.6.
> 
> Until you resize the dialog, then "magic" happens. The "magic" part is what I don't like. It is not discoverable and can't be observed by accessibility tools. As a user you have to "know" that something will happen or you don't and get surprised.

I have to disagree there since that is how everyone should expect a larger text field to work - what else should happen if you edit a multi line text area and press enter?

As before mentioned CTRL+ENTER works too (it always did). So if you want I could add it as initial info msg as in your version.

To me this is all surprisingly off topic. Since this PR restores original behavior while most discussion is about how to make the (new) multiline feature more discoverable.

Reply:
![v2](https://user-images.githubusercontent.com/114367/150639501-b8f475b5-926a-447d-860b-cc4314d2f995.png)
would this help?

Reply:
@mbien this doesn't restore the old behaviour. As Matthias says it adds "magic". So, we can restore the old behaviour by reverting the UI changes for multiline, make things clearer as Matthias' change does, or change the newline key to be `SHIFT+ENTER` or `ALT+ENTER` which is similar to other software.  I'd be happy if we got any of those 3 options in to NB13, but disagree on the behaviour in this PR.

Reply:
@neilcsmith-net
the screenshot above, option 1:
- says press CTRL+ENTER just like in Matthias'es version (just wanting to make sure that you didn't overlook it)
- it addresses the discoverability issue by suggesting to resize the window for multi line use
 
option 2:
 - add resize info text just like above for discoverability (but don't mention CTRL+ENTER)
 - always dispatch ENTER as dialog action
 - use SHIFT+ENTER as new line action as you suggested (and mention it in the msg)
 
I would prefer 1 since it has identical behavior to 12.6 and the hint msg clearly describes how to use the new feature. Option 2 is close second since it requires to use unintuitive SHIFT+ENTER for newline in text areas, but since its a niche feature, i would be ok with it.

I would also like to avoid making the textfields larger by default since it makes the dialog uglier for the most common usecase: single line search

Reply:
I'll leave you and @matthiasblaesing to decide over option 2 and his suggestion.  I agree with him that anything changing the behaviour based on window sizing is the wrong approach, with or without the tip.  Still somewhat prefer his suggestion.

Incidentally `SHIFT+ENTER` and `ALT+ENTER` are commons ways to handle this in chat programs and spreadsheets, so not that unintuitive, particularly as a retrofit.

Reply:
I would prefer option 2.

Reply:
changed it to option 2 and took the border change from @matthiasblaesing 's commit with slight modifications, since some LAFs use the outer components border instead of the editor's border. (e.g as visible on the screenshot https://github.com/apache/netbeans/pull/3489#discussion_r790134047, there would be no space between border and text and look inconsistent)